cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -817,6 +824,140 @@ def ceph_storage(ceph_admin):
     set_state('ceph-storage.configured')
 
 
+@when('config.changed.default-storage')
+def render_ceph():
+    remove_state('ceph-storage.configured')
+
+
+@when('ceph-storage.available')
+@when('cdk-addons.configured')
+@when_not('ceph-storage.configured')
+def ceph_csi_storage(ceph_admin):
```

Comment:
It looks like this handler will only run once.

What happens if the cdk-addons snap upgrades in the background, and the new version comes with new templates? The charm will be stuck on the old ones right? I don't think that's what we want.